### PR TITLE
Update logic for verified homeless history visibility

### PR DESCRIPTION
**Pull request by gigxz**
Updates to https://github.com/greenriver/hmis-warehouse/pull/2355 after QA

### CODE REVIEW

File: spec/models/grda_warehouse/client_file_spec.rb [hunk 1]
```diff
@@ -153,37 +153,44 @@
...
     let(:config) { create :config }
     let!(:verified_homeless_tag) { create :available_file_tag, name: 'Homeless Verification', verified_homeless_history: true }
-    let!(:file) { create :client_file }
-    let!(:own_file) { create :client_file, user: user }
+    # verified homeless history created by another user
+    let!(:history_file) { create :client_file }
+    # verified homeless history created by own user
+    let!(:own_history_file) { create :client_file, user: user, client: history_file.client }
+    # other type of file created by another user, for another client
+    let!(:other_file) { create :client_file }
```
Comment (gigxz) [inline spec/models/grda_warehouse/client_file_spec.rb:165]: I added another file to the spec (NOT a homeless history) to make sure that the `sql = sql.or(has_full_housing_release.and(is_not_verified_homeless_history))` pieces are working correctly.

File: app/models/grda_warehouse/client_file.rb [hunk 2]
```diff
@@ -47,39 +49,36 @@ class ClientFile < GrdaWarehouse::File
...
-        sql = is_own_file
-        sql = sql.or(is_consent_form) if GrdaWarehouse::Config.get(:consent_visible_to_all)
-        sql = sql.or(is_verified_homeless_history) if can_view_homeless_verification_pdfs
         where(sql)
       # You have specific permission to generate homeless verification PDFs
-      elsif user.can_generate_homeless_verification_pdfs?
+      elsif user.can_generate_homeless_verification_pdfs? && GrdaWarehouse::Config.get(:verified_homeless_history_visible_to_all)
         where(is_verified_homeless_history)
+      elsif user.can_generate_homeless_verification_pdfs?
+        where(is_own_file.and(is_verified_homeless_history))
```
Comment (gigxz) [inline app/models/grda_warehouse/client_file.rb:81]: I added this after we paired, @eanders. Before, users with `can_generate_homeless_verification_pdfs` could still only see their own files, even if `verified_homeless_history_visible_to_all` was set. I think we agreed that the "visible to all" setting should apply across the board. Let me know if I should change it back.

Comment (eanders) [reply app/models/grda_warehouse/client_file.rb:81]: this is perfect.

**Review by eanders — APPROVED** [review]
Assuming tests pass, this seems ready to me.  Thanks again for your tenacity on this one.

**Pull request merged**